### ISSUE TITLE
pipelines: pipeline progress value 0 for completed task `Bug`

### ISSUE THREAD

**Issue by e2corporation** [merico-dev/lake#977]
## Config UI / Pipelines / Jenkins
It appears that a completed task for the Jenkins Pipeline Data Provider (Plugin) does not correctly set `progress:1` for a task with a status of `TASK_COMPLETED`. Expected behavior that a completed task should have a progress value of `1`. Investigate other Plugins for similar behavior.

### Steps to Reproduce
1. Create a new Pipeline with **Jenkins** configuration.
2. Fetch Task Status
3. Observe that progress=0 for a completed task

## Pipeline `COLLECT 1639691068079`
```
{
  "ID": 279,
  "CreatedAt": "2021-12-16T22:56:09.913Z",
  "UpdatedAt": "2021-12-16T22:56:10.258Z",
  "name": "COLLECT 1639691068079",
  "tasks": [
    [
      {
        "plugin": "jenkins",
        "options": {}
      }
    ]
  ],
  "totalTasks": 1,
  "finishedTasks": 1,
  "beganAt": "2021-12-16T22:56:09.96Z",
  "finishedAt": "2021-12-16T22:56:10.257Z",
  "status": "TASK_COMPLETED",
  "message": "",
  "spentSeconds": 1
}
```
### Pipeline Tasks

```
{
  "count": 1,
  "tasks": [
    {
      "ID": 635,
      "CreatedAt": "2021-12-16T22:56:09.928Z",
      "UpdatedAt": "2021-12-16T22:56:10.248Z",
      "plugin": "jenkins",
      "options": {},
      "status": "TASK_COMPLETED",
      "message": "",
      "progress": 0, ⚠️ **Why is Progress == 0 when Task is Completed?**
      "pipelineId": 279,
      "pipelineRow": 1,
      "pipelineCol": 1,
      "beganAt": "2021-12-16T22:56:09.98Z",
      "finishedAt": "2021-12-16T22:56:10.247Z",
      "spentSeconds": 1
    }
  ]
}
```

**Comment by klesh** [comment]
Can you post devlake log for investigation?

**Comment by klesh** [comment]
And it might be fixed by PR #991, please verify

**Comment by e2corporation** [comment]
@klesh I think that spent seconds is only being updated when a task completes for fails, but it should be updated frequently especially for the RUNNING status. I will check again once latest changes from `main` are pulled.
